### PR TITLE
docs: add option tables, fix 27 inaccuracies, add API sync CI test

### LINKED ISSUE #342: docs: add option tables for builder and walker With* functions
## Context

The `internal/doctest/doc_test.go` test (PR #341) verifies that every exported `With*` function appears in its package's `deep_dive.md` option table. Two packages have large exception lists because they document options in prose/code examples rather than formal tables:

- **builder**: 30+ entries (tag, server, response, request body, operation, parameter validation, builder config options)
- **walker**: 20+ entries (handler registrations like `WithSchemaHandler`, `WithOperationHandler`, plus `WithFilePath`, `WithParsed`, `WithContext`)

Staleness checks prevent these lists from silently drifting, but the size means a significant fraction of public API surface bypasses the doc-check.

## Proposal

1. **builder/deep_dive.md** — Add option tables organized by option type:
   - `ParamOption` (validation constraints)
   - `TagOption` (description, external docs)
   - `ServerOption` (description, variables)
   - `ResponseOption` (content type, example, header, schema, ref)
   - `OperationOption` (handler, security, parameters)
   - Builder-level config (generic naming, schema field processor, validation)

2. **walker/deep_dive.md** — Add a handler registration table listing all `With*Handler` functions with their callback signatures.

3. **Shrink exception lists** — As tables are added, remove corresponding entries from `sourceExceptions` in `internal/doctest/doc_test.go`.

## Scope

Documentation-only changes (~100+ lines of markdown tables). No code changes needed.

### PR DESCRIPTION
**Pull request by erraggy**
## Summary

- Add comprehensive option tables to builder and walker deep_dive.md files (8 new builder tables, 2 new walker tables), removing 58 exception entries from doctest
- Fix 27 code example inaccuracies found via full audit across whitepaper, developer guide, README, and example READMEs (nonexistent functions, wrong signatures, internal package refs, deprecated usage)
- Add `TestDocCodeExampleAPISync` CI test that scans 93 markdown files and verifies every `package.Symbol` reference in Go code examples against AST-extracted symbol tables for all 12 public packages

Closes #342

### Option Tables (commit 1)
| Package | Tables Added |
|---------|-------------|
| builder | Tag, Server, Server Variable, Response Detail, Response-Level Operation, Request Body Detail, Advanced Operation, Parameter Validation, Generic Naming, Schema Field Processing |
| walker | Handler Registration (21 entries), Input Options |

### Inaccuracies Fixed (commit 2)
| Category | Count | Examples |
|----------|-------|---------|
| Nonexistent functions | 7 | `WithDocument`, `ValidateParsed`, `Build`, `ToYAML`, `ResponseConfig` |
| Wrong signatures | 5 | `WithParameter` positional args, `WithRequestBody` arg order, `WithCollisionHandlerFor` reversed |
| Internal package refs | 2 | `severity.SeverityCritical` → `differ.SeverityCritical` |
| Deprecated usage | 2 | `WithMaxDepth` → `WithMaxSchemaDepth` |
| Pointer/value mismatches | 2 | `WithParsed(result)` → `WithParsed(*result)` |
| Method vs function | 2 | `joiner.WriteResult()` → `j.WriteResult()` |
| Other | 7 | Variadic args, renamed constants, entire code block rewrites |

### CI Verification Test (commit 3)
- `TestDocCodeExampleAPISync` uses `go/ast` to build symbol tables for all 12 public packages
- Scans all user-facing markdown files for Go fenced code blocks
- Verifies every qualified reference (`pkg.Symbol`) exists as an exported symbol
- Flags internal package references (e.g., `severity.*`, `httputil.*`)
- Exception mechanism with staleness checks for intentional edge cases
- Adds 93 new subtests (one per markdown file)

## Test plan

- [x] `make check` passes (8498 tests)
- [x] `TestDocCodeExampleAPISync` passes with 93 subtests
- [x] `TestDeepDiveOptionTables` passes (58 fewer exceptions)
- [x] Security review: clean, no findings
- [x] Verify intentional breakage detected: introduced `WithNonexistent` in README.md, test caught both occurrences with file:line and clear message

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: examples/workflows/README.md [hunk 1]
```diff
@@ -176,8 +176,8 @@ All workflows demonstrate the parse-once pattern for maximum performance:
 parsed, _ := parser.ParseWithOptions(parser.WithFilePath("spec.yaml"))
 
 // Reuse for multiple operations
-fixer.FixWithOptions(fixer.WithParsed(parsed))
-validator.ValidateWithOptions(validator.WithParsed(parsed))
+fixer.FixWithOptions(fixer.WithParsed(*parsed))
+validator.ValidateWithOptions(validator.WithParsed(*parsed))
 ```
```
[the remark this answers is not in the document]
Comment (erraggy) [reply examples/workflows/README.md]: Intentional — this is a concise snippet demonstrating the parse-once pattern. The `parsed, _ :=` convention is standard across Go README examples for showing the happy path without error-handling boilerplate. Adding `if err != nil` would triple the snippet length and distract from the point (reuse the parse result).
